docs/gettinginvolved/plugins/building-driver.rst

```diff
@@ -0,0 +1,161 @@
+
+The best way to build your driver is to model it on the canonical implementations provided by the ClusterHQ team.
+These drivers include:
+
+* `OpenStack Cinder <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/cinder.py>`_
+* `Amazon EBS <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/ebs.py>`_
+
+After you have implemented the driver, you will need to test your implementation, and ClusterHQ provide a number of test suites to help you do this.
+Passing all tests means that your driver is ready to be deployed with Flocker in a customer environment.
+   
+Before beginning the testing steps below, you can review our :ref:`build-flocker-driver-faq` for common issues encountered during driver development and testing.
+
+Testing Your Driver
+===================
+
+#. Implement minimal functional tests:
+
+   To test that your implementation is correct you should instantiate a generic test suite that makes sure your class correctly implements the interface:
+
+   .. code-block:: python
+
+      from uuid import uuid4
+      from flocker.node.agents.test.test_blockdevice import make_istatechange_tests
+
+      def api_factory(test):
+          # Return an instance of your IBlockDeviceAPI implementation class, given
+          # a twisted.trial.unittest.TestCase instance.
+
+      # Smallest volume to create in tests, e.g. 1GiB:
+      MIN_ALLOCATION_SIZE = 1024 * 1024 * 1024
+
+      # Minimal unit of volume allocation, e.g. 1MiB:
+      MIN_ALLOCATION_UNIT = 1024 * 1024
+
+      class YourStorageTests(make_istatechange_tests(
+          api_factory, MIN_ALLOCATION_SIZE, MIN_ALLOCATION_UNIT,
+          # Factory for valid but unknown volume id specific to your backend:
+          lambda test: unicode(uuid4()))):
+          """
+          Tests for your storage.
+          """
+
+   If you wish the tests to cleanup volumes after each run, please provide a cleanup version of ``IBlockDeviceAPI``.
+   For an example of a clean up script, see the `EBS API with cleanup <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/test/blockdevicefactory.py>`_ inside ``api_factory``.
+
+   You can run these tests with the ``trial`` test runner, provided by `Twisted <(http://twistedmatrix.com/trac/wiki/TwistedTrial)>`_, one of Flocker's dependencies:
+
+   .. prompt:: bash $
+
+      trial yourstorage.test_yourstorage
+
+#. Additional functional tests:
+
+   You are encouraged to write additional functional tests to cover logic specific to your driver implementation.
+   For example, here are some `EBS driver-specific tests <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/functional/test_ebs.py>`_ written by ClusterHQ.
+
+#. Run acceptance tests:
+
+   After all functional tests pass, please run acceptance tests according to our :ref:`acceptance-testing` documentation.
+
+#. Setup Continuous Integration environment for tests.
```

Comment:
I think that this should be "Set up a Continuous Integration environment for tests"